apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/AuthenticationProviderActivity.java

```diff
@@ -93,9 +111,65 @@ public void onClick(View v) {
                 }
             }
         }));
-        ListView listView = findViewById(R.id.list);
-        listView.setAdapter(new TestFeaturesListAdapter(featureList));
-        listView.setOnItemClickListener(TestFeatures.getOnItemClickListener());
+        mListView = findViewById(R.id.list);
+        loadAuthStatus(sUserInformation == null);
+        mListView.setOnItemClickListener(TestFeatures.getOnItemClickListener());
+    }
+
+    private void loadAuthStatus(boolean _default) {
+        if (mAuthInfoTestFeature != null) {
+            mFeatureList.remove(mAuthInfoTestFeature);
+        }
+        mAuthInfoTestFeature = getDefaultAuthenticationTestFeature();
+        if (!_default) {
+            if (isAuthenticated()) {
+                mAuthInfoTestFeature = getAuthenticatedTestFeature();
```

Comment:
Nitpick: let's make it a bit readable
```suggestion
                mAuthInfoTestFeature = isAuthenticated() ? getAuthenticatedTestFeature() : getNotAuthenticatedTestFeature();
```

Reply:
I guess not applicable anymore